libpod/container_top_linux.go

```diff
@@ -297,19 +298,25 @@ func (c *Container) execPS(psArgs []string) ([]string, bool, error) {
 	if err != nil {
 		return nil, false, err
 	}
-	defer wPipe.Close()
 	defer rPipe.Close()
 
+	wg := &sync.WaitGroup{}
```

Comment:
I think we can use an error channel instead and thereby also return the `scanner.Err()` instead of logging it.

Reply:
yeah that should be better as exit 0 with missing output would be a bad idea, and logrus is not seen on the remote client so I will fix that up.